plugins/org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/workspace/EclipseProjectConfig.xtend

```diff
@@ -63,6 +69,42 @@ class EclipseProjectConfig implements IProjectConfig {
 	override findSourceFolderContaining(URI member) {
 		sourceFolders.findFirst[folder|folder.path.isPrefixOf(member)]
 	}
+	
+	override getWorkspaceConfig() {
+		new EclipseWorkspaceConfig(project.workspace.root, projectConfigProvider)
+	}
+	
+}
+
+@Data
+class EclipseWorkspaceConfig implements IWorkspaceConfig {
+	
+	IWorkspaceRoot workspaceRoot
+	val EclipseProjectConfigProvider projectConfigProvider
+	
+	override Set<? extends EclipseProjectConfig> getProjects() {
+		workspaceRoot.projects.filter[project.exists].map[projectConfigProvider.createProjectConfig(project)].toSet
```

Comment:
should this be project.isAccessible instead of exists? Otherwise we'll return closed projects, too?